tests/metrics/test_relation_argument_distance_collector.py

```diff
@@ -0,0 +1,240 @@
+import dataclasses
+
+import pytest
+from pytorch_ie import Annotation, Document
+from pytorch_ie.annotations import BinaryRelation, LabeledSpan, NaryRelation
+from pytorch_ie.core import AnnotationList, annotation_field
+from pytorch_ie.documents import TextBasedDocument, TokenBasedDocument
+
+from pie_modules.metrics import RelationArgumentDistanceCollector
+
+
+@dataclasses.dataclass
+class TestDocument(TextBasedDocument):
+    entities: AnnotationList[LabeledSpan] = annotation_field(target="text")
+    relations: AnnotationList[BinaryRelation] = annotation_field(target="entities")
+
+
+def test_relation_argument_distance_collector():
+    doc = TestDocument(
+        text="This is the first entity. This is the second entity. And, this is the third entity."
+    )
+    doc.entities.append(LabeledSpan(start=0, end=25, label="entity"))
+    doc.entities.append(LabeledSpan(start=26, end=52, label="entity"))
+    doc.entities.append(LabeledSpan(start=53, end=83, label="entity"))
+    doc.relations.append(
+        BinaryRelation(head=doc.entities[0], tail=doc.entities[1], label="relation_label_1")
+    )
+    doc.relations.append(
+        BinaryRelation(head=doc.entities[1], tail=doc.entities[2], label="relation_label_2")
+    )
+
+    statistic = RelationArgumentDistanceCollector(layer="relations")
+    values = statistic(doc)
+    assert values == {
+        "ALL": {"len": 4, "mean": 54.5, "std": 2.5, "min": 52.0, "max": 57.0},
+        "relation_label_1": {"len": 2, "mean": 52.0, "std": 0.0, "min": 52.0, "max": 52.0},
+        "relation_label_2": {"len": 2, "mean": 57.0, "std": 0.0, "min": 57.0, "max": 57.0},
+    }
+
+
+def test_relation_argument_distance_collector_with_n_ary_relation():
+    doc = TestDocument(
+        text="This is the first entity. This is the second entity. And, this is the third entity."
+    )
+    doc.entities.append(LabeledSpan(start=0, end=25, label="entity"))
+    doc.entities.append(LabeledSpan(start=26, end=52, label="entity"))
+    doc.entities.append(LabeledSpan(start=53, end=83, label="entity"))
+    doc.relations.append(
+        NaryRelation(
+            arguments=[
+                doc.entities[0],
+                doc.entities[1],
+            ],
+            roles=["head", "tail"],
+            label="relation_label_1",
+        )
+    )
+    doc.relations.append(
+        NaryRelation(
+            arguments=[
+                doc.entities[1],
+                doc.entities[2],
+            ],
+            roles=["head", "tail"],
+            label="relation_label_2",
+        )
+    )

```

Comment:
Can you use a single `NaryRelation`, but with 3 arguments? For `NaryRelation` we handle the distance by calculating over all combinations of args, which would be the same when having two args compared to `BinaryRelations` (so already more or less covered by the other test), but different when having more args.